### ISSUE TITLE
Abusive non-exhaustive error

### ISSUE THREAD

**Issue by GuillaumeGomez** [rust-lang/rust#35859]
This code:

```rust
enum AnotherRes<T> {
    Success(T),
    Failure,
}

fn main() {
    let x: Result<AnotherRes<i32>, ()> = Ok(AnotherRes::Success(1i32));

    match x {
        Ok(AnotherRes::Success(v)) => println!("good! {}", v),
        Err(_) => {}
    }
}
```

Throws the following error:

```bash
error: non-exhaustive patterns: `Ok(Failure)` not covered [--explain E0004]
 --> <anon>:9:5
  |>
9 |>     match x {
  |>     ^
```

However, all patterns are covered so this error shouldn't be reported.

**Comment by TimNN** [comment]
@GuillaumeGomez: How is `Ok(Failure)` covered? It's not as far as I can tell.

**Comment by nagisa** [comment]
…abusive?

**Comment by arielb1** [comment]
Not a bug.

**Comment by GuillaumeGomez** [comment]
@arielb1: No it's not! `Err(_)` patterns should cover all remaining possibilities.
